test/built-ins/Promise/any/invoke-resolve-on-promises-every-iteration-of-custom.js

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2019 Sergey Rubanov. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+  Invocation of the constructor's `resolve` method for iterable with promise values
+esid: sec-promise.any
+info: |
+  5. Let result be PerformPromiseAny(iteratorRecord, C, promiseCapability).
+
+  Runtime Semantics: PerformPromiseAny
+
+  8. Repeat
+    ...
+    i. Let nextPromise be ? Call(promiseResolve, constructor, « nextValue »).
+
+flags: [async]
+features: [Promise.any]
+---*/
+class Custom extends Promise {}
```

Comment:
This test, while technically correct, bugs me a lot as it doesn't need a lot of things being used here. To start is that Custom can be just a simple constructor and there is no need to inherit from Promise at all.

Then, the usage of new instances (_from lines 23 to 25_) doesn't seem to be what we are testing here.

For the test description, I want to know that the given `this` value is used to get the property `resolve` to be called. There we can observe calls, args, and return anything we need.

Reply:
> I want to know that the given this value is used to get the property resolve to be called.

That's tested in another file.